### PR TITLE
[FEATURE: UIPT-30] - Wish list page

### PR DESCRIPTION
**Pull request by IrynaMukhina**

### CODE REVIEW

File: src/app/components/wish-list/wish-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+})
+export class WishListComponent implements OnInit, OnDestroy {
+  public getProductsSub: Subscription;
+  public productData: Array<IProductShortInfo> = [];
+  liked$: Observable<Array<string>>;
+  likedArray: Array<string>;
+
+  constructor(
+    private productService: ProductService,
+    private store: Store<IAppState>) { }
```
Comment (AlexLarusso) [inline src/app/components/wish-list/wish-list.component.ts]: Please align:

```
 constructor(
    private productService: ProductService,
    private store: Store<IAppState>
) { }
```

File: src/app/shared/wishlist-button/wish-list-button.component.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit, OnDestroy, Input } from '@angular/core';
```
Comment (AlexLarusso) [inline src/app/shared/wishlist-button/wish-list-button.component.ts]: I think it is not shared component. 
It's called `wish-list-button`, so I guess, it is part of wish-list component. Move to wish-list/components

File: src/app/shared/wishlist-button/wish-list-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  constructor(private store: Store<IAppState>) { }
+
+  public ngOnInit(): void {
+    const localStorageLiked = JSON.parse(localStorage.getItem('liked'));
+
+    if (localStorageLiked) {
+      this.store.dispatch(new SetToWishList(localStorageLiked));
+      this.isLiked = localStorageLiked.includes(this.productId);
+    }
```
[the remark this answers is not in the document]
Comment (AlexLarusso) [reply src/app/shared/wishlist-button/wish-list-button.component.ts]: I think you should use selectors here, and SetToWishList one time

File: src/app/store/reducers/wish-list.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  likedProducts: Array<string>;
+}
+
+const initialState = {
+  likedProducts: []
+};
+
+export function wishListReducer(state = initialState, action: WishListActions): IState {
+  const likedProductsArray = state.likedProducts;
+
```
Comment (AlexLarusso) [inline src/app/store/reducers/wish-list.reducer.ts]: Use destructoring:
```
const { type, payload } = action;

File: src/app/components/wish-list/wish-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    this.liked$ = this.store.select(getLiked);
+    this.liked$.subscribe(data => {
+      this.likedArray = data;
+      this.productData = this.productData.filter(el => data.includes(el.productId.toString()));
+    });
+
+    this.getProductsSub = this.productService.getProducts(ProductFormat.short)
+    .subscribe(data => this.productData = data.filter((el) =>
+      this.likedArray.includes(el.productId.toString()
+    )));
```
[the remark this answers is not in the document]
Comment (AlexLarusso) [reply src/app/components/wish-list/wish-list.component.ts]: My mistake. This code is OK. Sorry.

[the remark this answers is not in the document]
Comment (IrynaMukhina) [reply src/app/components/wish-list/wish-list.component.ts]: I have already rewrite it with using switchMap and it's really looks better =)

File: src/app/interfaces/product-details.interface.ts [hunk 6]
```diff
@@ -10,4 +10,5 @@ export interface IProductDetails {
   season: string | Array<string>;
   options: IProductOptions;
   description: IProductDescription;
+  productId: string;
```
Comment (AlexLarusso) [inline src/app/interfaces/product-details.interface.ts]: `productId` should be number.

File: src/app/shared/product-item/product-item.component.ts [hunk 7]
```diff
@@ -13,8 +9,5 @@ export class ProductItemComponent {
   @Input() public imgUrl = '../../../assets/server-data/images/image-not-found.png';
   @Input() public productTitle = 'Product Title';
   @Input() public productPrice = 0;
-  @Input() public productId = 1;
-
-  public faHeart = faHeart;
-  public product: IProduct;
+  @Input() public productId: string;
```
Comment (AlexLarusso) [inline src/app/shared/product-item/product-item.component.ts]: number should be

File: src/app/shared/wish-button/wish-button.component.ts [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    this.isLiked = !this.isLiked;
+
+    this.isLiked
+      ? this.store.dispatch(new AddToWishList(this.productId))
+      : this.store.dispatch(new RemoveFromWishList(this.productId));
+
+    this.setAllToLocalStorage();
+  }
+
+  public setAllToLocalStorage(): void {
```
Comment (AlexLarusso) [inline src/app/shared/wish-button/wish-button.component.ts]: I think it is private. 
Please check every function in your PR.

If you have questions when use private and when public - please ask

File: src/app/components/wish-list/wish-list.component.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        this.likedArray = likedProducts;
+
+        return this.getProducts.pipe(
+          map(products => {
+            this.productData = products;
+            this.productData = this.productData.filter(el => this.likedArray.includes(el.productId));
+          })
+        );
+      }
+    )).subscribe();
```
[the remark this answers is not in the document]
Comment (IrynaMukhina) [reply src/app/components/wish-list/wish-list.component.ts]: I have replaced some logic inside it. Now it isn't empty.

File: src/app/interfaces/product-details.interface.ts (hunk 6 above)
Comment (IrynaMukhina) [reply src/app/interfaces/product-details.interface.ts]: it's string:
image

**Comment by IrynaMukhina** [comment]
@AlexLarusso thank you very much for reviewing!

File: src/app/interfaces/product-details.interface.ts (hunk 6 above)
Comment (AlexLarusso) [reply src/app/interfaces/product-details.interface.ts]: OK. FYI: But in most cases id should be number.

**Pull request merged**